src/commands/demo-store.js

```diff
@@ -0,0 +1,361 @@
+      this.exit(1)
+    })
+  }
+
+  /**
+   * Display a formatted message indicating the given store has been downloaded properly
+   *
+   * @param {object} manifest The "manifest" for the store that is being downloaded
+   */
+  displayCompletion(manifest) {
+    this.log(`${chalk.bold(manifest.name)}
+
+${chalk.dim(manifest.description)}`)
+  }
+
+  /**
+   * Attempt to run the example store that was downloaded by installing dependencies with NPM or Yarn and running
+   * any additional configued commands
+   *
+   * @param {object} manifest The "manifest" for the store that is being downloaded
+   * @return {Promise} Resolves when NPM installation (if applicable) is complete and further commands have finished
+   */
+  async runExample(manifest) {
+    const {npm, buildScripts} = manifest
+
+    try {
+      await this.writeEnv(manifest)
+    } catch (error) {
+      return
+    }
+
+    // Don't continue if there's no NPM configuration
+    if (!npm) {
+      return
+    }
+
+    // Allow "yarn" to be used
+    const command = npm === 'yarn' ? 'yarn' : 'npm'
+    // Run the commands in the destination directory
+    const cwd = this.getDestinationDirectory()
+    // Prep the args that will be added to the run command
+    const baseArgs = npm === 'yarn' ? ['--cwd', cwd] : ['--prefix', cwd]
+
+    const spinner = ora({
+      text: 'Installing NPM dependencies...',
+      stream: process.stdout,
+    }).start()
+
+    // Spawn a subprocess for dependency installation
+    await spawnPromise.spawn(command, [...baseArgs, 'install'])
+    spinner.stop()
+
+    // Loop through additional scripts and await their execution
+    buildScripts.forEach(async script => {
+      await spawnPromise.spawn(command, [...baseArgs, 'run', script], {stdio: 'inherit'})
+    })
+  }
+
+  async writeEnv(manifest) {
+    const {flags: {env}} = this.parse(DemoStoreCommand)
+
+    let options = manifest.dotenv || {}
+
+    if (Array.isArray(env)) {
+      env.forEach(envDefinition => {
+        const matches = envDefinition.match(/^([\w\d_]+)=(.+)$/i)
+        if (!matches) {
+          this.log(chalk.red(`Could not understand env definition: ${envDefinition}. Please provide env settings as key=value. Eg. --env key=value`))
+          return
+        }
+
+        options[matches[1]] = matches[2]
+      })
+    }
+
+    // Check for "dotenv" settings and update/write a .env file
+    if (typeof manifest.dotenv === 'object') {
+      const env = await envWriter.create(`${this.getDestinationDirectory(manifest)}${sep}.env`)
+
+      Object.entries(manifest.dotenv).forEach(([key, value]) => {
+        // Check for an injection point of chec keys
+        const matches = typeof value === 'string' && value.match(/^%chec_([ps])key%$/)
+
+        if (!matches) {
+          env.set(key, value)
+          return
+        }
+
+        if (!Auth.isLoggedIn()) {
+          this.log(`${chalk.bgRed.white('Could not run the example store. You must be logged in')}
```

Comment:
This kicks in pretty late in the process:

```
$ bin/run demo-store
? Please choose a demo store to install: Shoe store (react-shoe-store): An example
 shoe store showcasing prod...
Downloading from https://github.com/ScopeyNZ/example.checkout.v2/archive/react.js.zip
✔ Downloaded react-shoe-store to /Users/robertaverill/dev/chec/chec-cli/react-shoe-store
Shoe store

An example shoe store showcasing product variants and hosted checkout
Could not run the example store. You must be logged in

This store requires a .env file with your Chec.io public key provided as "REACT_APP_COMMERCEJS_PUBLIC_KEY"
```

I raised an issue at #12 which might make this redundant, maybe add a todo comment so we don't forget to remove this logic when #12 gets done?

Reply:
Yeah I wasn't sure about this one 😅 . I figured it would be technically possible to have an example that doesn't actually require your keys - not sure what context that would be... I can just move the check up to make it easier.